src/GraphQL/Server.hs

```diff
@@ -112,34 +132,40 @@ instance Defaultable Bool
 
 instance Defaultable Text
 
+instance Defaultable (Maybe a) where
+  -- | The default for @Maybe a@ is @Nothing@.
+  defaultFor _ = pure Nothing
+
 -- TODO not super hot on individual values having to be instances of
 -- HasGraph but not sure how else we can nest either types or
 -- (Object _ _ fields). Maybe instead of field we need a "SubObject"?
-instance forall m. (MonadThrow m, MonadIO m) => HasGraph m Int32 where
+instance forall m. (Monad m) => HasGraph m Int32 where
```

Comment:
Pretty sure you don't need `Monad` constraint here, or anywhere for the scalar types.